src/paginationCalls/pageDescriptor.ts

```diff
@@ -143,7 +143,7 @@ export class PageDescriptor implements Descriptor {
     const asyncIterable = {
       [Symbol.asyncIterator]() {
         let nextPageRequest: RequestType | null | undefined = request;
-        const cache: {}[] = [];
+        const cache: Array<ResponseType | [string, ResponseType]> = [];
```

Comment:
Another option
`const cache: (ResponseType | [string, ResponseType]) = []`,
Base on my search `Array<T>` vs `T[]` are quite similar, [] is shorthand way, I choose `Array<T>` because it is more readable with multiple types in my opinion.
Let me know if you have any concern. @alexander-fenster Thanks.

Reply:
They are similar, and `gts` forces us to use `Array<>` for complex types and `[]` for simple types.  You did it correctly here.